src/Http/Http.Results/src/Results.cs

```diff
@@ -495,61 +495,61 @@ public static IResult RedirectToRoute(string? routeName = null, object? routeVal
     /// <param name="statusCode">The status code to set on the response.</param>
     /// <returns>The created <see cref="StatusCodeHttpResult"/> object for the response.</returns>
     public static IResult StatusCode(int statusCode)
-        => new StatusCodeHttpResult(statusCode);
+        => ResultsCache.StatusCode(statusCode);
 
     /// <summary>
     /// Produces a <see cref="StatusCodes.Status404NotFound"/> response.
     /// </summary>
     /// <param name="value">The value to be included in the HTTP response body.</param>
     /// <returns>The created <see cref="IResult"/> for the response.</returns>
     public static IResult NotFound(object? value = null)
-        => new NotFoundObjectHttpResult(value);
+        => value is null ? ResultsCache.NotFound : new NotFoundObjectHttpResult(value);
```

Comment:
Is there a follow up plan to use object pooling for the object overloads?

Reply:
Nope.